src/Compilers/CSharp/Portable/BoundTree/BoundQueryClause.cs

```diff
@@ -49,7 +49,7 @@ internal partial class BoundQueryClause
             BoundExpression unoptimizedForm,
             TypeSymbol type)
         {
-            if (value != this.Value || definedSymbol != this.DefinedSymbol || queryInvocation != this.Operation || castInvocation != this.Cast || binder != this.Binder || unoptimizedForm != this.UnoptimizedForm || type != this.Type)
+            if (value != this.Value || definedSymbol != this.DefinedSymbol || queryInvocation != this.Operation || castInvocation != this.Cast || binder != this.Binder || unoptimizedForm != this.UnoptimizedForm || !TypeSymbol.Equals(type, this.Type, TypeCompareKind.ConsiderEverything2))
```

Comment:
>ConsiderEverything2 [](start = 266, length = 19)

Update methods should use ```ConsiderEverything```

Reply:
This is not a generated method. Let's reserve this fine-tuning for follow-ups.

---
In reply to: [242689970](https://github.com/dotnet/roslyn/pull/31814#discussion_r242689970) [](ancestors = 242689970)